packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/index.tsx

```diff
@@ -138,19 +138,28 @@ const DeviceItem = ({ device, instances, closeModalApp, backgroundRoute }: Props
         (backgroundRoute.app === 'wallet' || backgroundRoute.app === 'dashboard');
     const instancesWithState = instances.filter(i => i.state);
 
-    const selectDeviceInstance = async (instance: Props['device']) => {
-        selectDevice(instance);
+    const gotoIndexIfNecessary = () => {
```

Comment:
I have struggled with naming 🤦  any better suggestions are welcome

Reply:
That's always the hardest part. I am okay with that. However, if I had this task, I would probably name it `handleRouting`, `handleIndexRouting`, `handleOptionalIndexRouting` 🤷‍♂️